### PR TITLE
refactor: cssinjs for image

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     image组件改为 cssinjs      |
| 🇨🇳 Chinese |    cssinjs for image       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
title需要改一下

**Comment by heiyu4585** [comment]
> title需要改一下

done

File: components/_util/theme/index.tsx [hunk 1]
```diff
@@ -201,13 +206,17 @@ export interface DerivativeToken extends ColorPalettes, Omit<DesignToken, 'durat
   tmpWarningColorDeprecatedBorder: string;
   tmpErrorColorDeprecatedBorder: string;
   tmpInfoColorDeprecatedBorder: string;
+
+  imagePreviewOperationColor: string;
```
Comment (zombieJ) [inline components/_util/theme/index.tsx:210]: 不透出组件级别 token，加个 `tmpXXX` 前缀，下同

File: components/image/index.tsx [hunk 2]
```diff
@@ -43,7 +47,14 @@ const Image: CompositionImage<ImageProps> = ({
     };
   }, [preview, imageLocale]);
 
-  return <RcImage prefixCls={prefixCls} preview={mergedPreview} {...otherProps} />;
+  return wrapSSR(
+    <RcImage
+      prefixCls={`${prefixCls}`}
+      preview={mergedPreview}
+      rootClassName={hashId}
```
Comment (zombieJ) [inline components/image/index.tsx:54]: 这边会把用户的 rootClassName 覆盖掉，需要合并一下。下同

File: components/image/style/index.tsx [hunk 3]
```diff
@@ -1,2 +1,327 @@
...
+  const { iconPrefixCls, white, black, animationDurationSlow, paddingXXS, marginXXS, prefixCls } =
+    token;
+  return {
+    position: 'absolute',
+    inset: 0,
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    color: white,
+    background: new TinyColor(black).setAlpha(0.5).toRgbString(),
```
Comment (zombieJ) [inline components/image/style/index.tsx:53]: hardcode 颜色的地方加一个 `// FIXME` 标注一下

File: components/image/style/index.tsx [hunk 4]
```diff
@@ -1,2 +1,327 @@
...
+    insetBlockStart: 0,
+    insetInlineEnd: 0,
+    zIndex: 1,
+    display: 'flex',
+    flexDirection: 'row-reverse',
+    alignItems: 'center',
+    width: '100%',
+    color: imagePreviewOperationColor,
+    listStyle: 'none',
+    background: new TinyColor(modalMaskBg).setAlpha(0.1).toRgbString(),
```
Comment (zombieJ) [inline components/image/style/index.tsx:91]: hardcode 颜色的地方加一个 `// FIXME` 标注一下

File: components/image/style/index.tsx [hunk 5]
```diff
@@ -1,2 +1,327 @@
...
+      overflow: 'hidden',
+    },
+
+    [`.${previewPrefixCls}-img`]: {
+      maxWidth: '100%',
+      maxHeight: '100%',
+      verticalAlign: 'middle',
+      transform: 'scale3d(1, 1, 1)',
+      cursor: 'grab',
+      transition: `transform 0.3s ${easeOut} 0s`,
```
Comment (zombieJ) [inline components/image/style/index.tsx:173]: 0.3s 可以替换成 `token.duration`

Comment (zombieJ) [reply components/image/style/index.tsx:173]: 下同

File: components/image/style/index.tsx [hunk 6]
```diff
@@ -1,2 +1,327 @@
...
+): UseComponentStyleResult {
+  const [theme, token, hashId] = useToken();
+
+  const inputToken: ImageToken = {
+    ...token,
+    prefixCls,
+    iconPrefixCls,
+    previewPrefixCls: `${prefixCls}-preview`,
+
+    animationDurationSlow: '0.3s', // FIXME: hard code in v4
```
Comment (zombieJ) [inline components/image/style/index.tsx:305]: 这边已经有对应 token 可以直接使用

File: components/image/style/index.tsx [hunk 7]
```diff
@@ -1,2 +1,327 @@
...
+    imageSizeBase: 48,
```
Comment (zombieJ) [inline components/image/style/index.tsx:306]: hardcode 数字都加一个 `// FIXME` 标注一下~

Comment (zombieJ) [reply components/image/style/index.tsx:306]: 下同

**Pull request closed without merging**